src/lightning/data/streaming/dataset.py

```diff
@@ -97,13 +97,11 @@ def __init__(
         self._state_dict: Optional[Dict[str, Any]] = None
 
     def _create_cache(self, worker_env: _WorkerEnv) -> Cache:
-        env = Environment(dist_env=self.distributed_env, worker_env=worker_env)
-
         if _should_replace_path(self.input_dir.path):
             # FIXME: Remove the `shard_rank` from the cache_path to enable reloading chunks for the second epoch
             # without paying the cost of re-download
             cache_path = _try_create_cache_dir(
-                input_dir=self.input_dir.path if self.input_dir.path else self.input_dir.url, shard_rank=env.shard_rank
```

Comment:
If we don't pass this shard rank, how will you prevent race conditions when workers download the same file (when they have indices from the same chunk)?

Reply:
We are using filelock to synchronise the processes.